### PR TITLE
Add option to override port in SNI DFP

### PR DESCRIPTION
**Pull request by ohadvano**
Adds an ability to dynamically set port value for the SNI DFP filter, by setting a filter state value for a per-connection stream.
This PR follows up #25234 that adds an option to dynamically set host value for SNI DFP.

### CODE REVIEW

File: test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_test.cc [hunk 1]
```diff
@@ -99,14 +105,14 @@ TEST_F(SniDynamicProxyFilterTest, LoadDnsCache) {
 
 TEST_F(SniDynamicProxyFilterTest, LoadDnsCacheFromFilterState) {
```
Comment (lizan) [inline test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_test.cc:106]: Add a test to make sure we fallback on the configured port when the `"envoy.upstream.dynamic_host"` is not set?

File: source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc [hunk 2]
```diff
@@ -54,9 +55,20 @@ Network::FilterStatus ProxyFilter::onNewConnection() {
...
-  uint32_t default_port = config_->port();
+  const StreamInfo::UInt32Accessor* dynamic_port_filter_state =
+      read_callbacks_->connection()
+          .streamInfo()
+          .filterState()
+          ->getDataReadOnly<StreamInfo::UInt32Accessor>("envoy.upstream.dynamic_port");
+
+  uint32_t port;
+  if (dynamic_port_filter_state) {
+    port = dynamic_port_filter_state->value();
```
Comment (lizan) [inline source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc:66]: check whether the port is at right range? for config this is done via PGV.

File: test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_test.cc (hunk 1 above)
Comment (ohadvano) [reply test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_test.cc]: done

File: source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc (hunk 2 above)
Comment (ohadvano) [reply source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc:67]: Yeah good catch, updated the code to use the default configured port if the port is out of range, and updated the docs accordingly

**Review by lizan — APPROVED** [review]

**Comment by lizan** [comment]
@ohadvano can you resolve conflicts so I can merge?

**Review by lizan — APPROVED** [review]

**Pull request merged**